### DISCUSSION TITLE
dbt deps git package differs from hub package

### DISCUSSION THREAD

**Discussion by RobbertDM** [dbt-labs/dbt-core#7225] [General]
Hey there,

I wanted to install 
```
  - package: EqualExperts/dbt_unit_testing
    version: 0.2.7
```
But that kept giving me the following error:
`Found duplicate project "dbt_utils".[...]`

Apparently, the fix was that I had
```yaml
  - git: "https://github.com/dbt-labs/dbt-utils.git"
    revision: 1.0.0
```
But apparently that's seen as a different package then
```yaml
  - package: dbt-labs/dbt_utils
    version: 1.0.0
```
When I swapped the first for the latter, everything installed
(cf. https://github.com/EqualExperts/dbt-unit-testing/issues/127)

Now, why I started this discussion is because I think we can do something better from a developer experience perspective.
Some possibilities:
* State in the [docs](https://docs.getdbt.com/docs/build/packages) that installing a git package is not the same as installing its equivalent hub package
* Include a warning about this in the `Found duplicate project [...]` warning
* Include the two sources of the duplicate project in the `Found duplicate project [...]` warning
* State in the docs that hub packages are preferred above git packages
* Resolve dependencies in a smarter way based on namespace, version, but not exact URL?

**Comment by jtcohen6** [comment]
Sorry for the confusion @RobbertDM! Glad you were able to get this sorted.

For what it's worth, we do have a note about this [in the docs](https://docs.getdbt.com/docs/build/packages#hub-packages-recommended) already:

> Where possible, we recommend installing packages via dbt Hub, since this allows dbt to handle duplicate dependencies. This is helpful in situations such as:
> 
> - Your project uses both the dbt-utils and Snowplow packages; and the Snowplow package also uses the dbt-utils package.
> - Your project uses both the Snowplow and Stripe packages, both of which use the dbt-utils package.
> 
> In comparison, other package installation methods are unable to handle the duplicate dbt-utils package.

**Reply by RobbertDM** [reply]
Ah yes, I need to learn to read more thoroughly. Thanks!
